File: Bit Manipulation/Lone-Integer.py
class Solution:
  def solve(self, nums):
    bits_frequency = [0 for bit in range(30)]
    for num in nums:
      for bit in range(30):
        if num & (1 << bit):
          bits_frequency[bit] += 1

    unique = 0
    for bit in range(30):
      if bits_frequency[bit] % 3 == 1:
        unique |= 1 << bit
    return unique

File: Bit Manipulation/Lone-Integer.cpp
int solve(std::vector<int>& nums) {
  std::vector<int> bits_frequency(30);
  for (int num : nums) {
    for (int bit = 0; bit < 30; ++bit) {
      if (num & (1 << bit)) {
        ++bits_frequency[bit];
      }
    }
  }

  int unique = 0;
  for (int bit = 0; bit < 30; ++bit) {
    if (bits_frequency[bit] % 3 == 1) {
      unique |= 1 << bit;
    }
  }
  return unique;
}
